File: BaekjoonAlgorithm/BA_1644/BJAlgo_1644.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;

public class BJAlgo_1644 {
    private static int[] primeNum;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        findPrimeNum(n);

        int low = 0, high = 0;
        int sum = primeNum[0];
        int result = 0;

        while (low <= high && high < primeNum.length - 1) {
            if (sum < n) {
                sum += primeNum[++high];
            } else if (sum == n) {
                result += 1;
                sum += primeNum[++high];
            } else if (sum > n) {
                sum -= primeNum[low++];
            }
        }

        if (primeNum[primeNum.length - 1] == n) {
            result += 1;
        }

        System.out.println(result);
    }

    private static void findPrimeNum(int n) {
        boolean[] isPrimeNum = new boolean[n + 1];
        Arrays.fill(isPrimeNum, true);

        ArrayList<Integer> primeNumList = new ArrayList<>();

        for (int i = 2; i < n + 1; i++) {
            if (isPrimeNum[i]) {
                int j = 2;
                while (i * j < n + 1) {
                    isPrimeNum[i * j] = false;
                    j += 1;
                }
                primeNumList.add(i);
            }
        }

        primeNum = primeNumList.stream().mapToInt(i -> i).toArray();
        if (primeNum.length == 0) {
            primeNum = new int[1];
        }
    }
}

File: BaekjoonAlgorithm/202102/BA_1644/BJAlgo_1644.py
from sys import stdin


def find_prime_num():
    is_prime_num = [True] * (n + 1)

    for i in range(2, n + 1):
        if is_prime_num[i]:
            end = (n + 1) // i if (n + 1) % i == 0 else (n + 1) // i + 1
            for j in range(2, end):
                is_prime_num[i * j] = False
            prime_num.append(i)

    if len(prime_num) == 0:
        prime_num.append(0)


n = int(stdin.readline())

prime_num = []
find_prime_num()

print(prime_num)

low = 0
high = 0
s = prime_num[0]
result = 0

while low <= high < len(prime_num) - 1:
    if s < n:
        high += 1
        s += prime_num[high]
    elif s == n:
        result += 1
        high += 1
        s += prime_num[high]
    elif s > n:
        s -= prime_num[low]
        low += 1

if prime_num[-1] == n:
    result += 1

print(result)
